packages/core/src/properties/root.ts

```diff
@@ -68,6 +68,8 @@ export class Root extends Property {
 	@GraphChildList private skins: Link<Root, Skin>[] = [];
 	@GraphChildList private textures: Link<Root, Texture>[] = [];
 
+	private scene?: number;
```

Comment:
Let's make this a `@GraphChild` property, similar to this example:

https://github.com/donmccurdy/glTF-Transform/blob/7be7f9ddaab098af79b1eea3b67adac3d9c453ed/packages/core/src/properties/skin.ts#L22

The advantage of this is that if the scene is disposed, or the scene order changes, the references are maintained. For example:

```ts
root.setDefaultScene(scene);
root.getDefaultScene(); // -> scene

scene.dispose();

root.getDefaultScene(); // -> null
```

Reply:
Ah, make sense, I didn't fully understand the whole GraphChild and Link logic and was lazily trying to skip that part in the hope it wouldn't be necessary :D.

Just to clarify, you would like to define a `scene` attribute that is a link/reference to a Scene object in the `scenes` list, so that the actual value is determined at runtime when writing the json for example?
In that case, would I need a `sceneIndexMap`  similar to the `nodeIndexMap` in the writer context?
Also, once the `root.scene` is not a number anymore, would the setDefaultScene of the document and root accept a Scene reference or a number (and then map the number to a Scene)?
For consistency with other methods seems like I should use the former.